lua/better-breachable-doors/sv_breach.lua

```diff
@@ -0,0 +1,757 @@
+
+        if hit then
+            collidingPlayers[ #collidingPlayers + 1 ] = ply
+        end
+    end
+
+    return collidingPlayers
+end
+
+-- Called regularly to check if any player is colliding with any door that is respawning.
+BBD.CheckPlayerCollisions = function()
+    local time = CurTime()
+
+    -- Don't check too often
+    if BBD.LastCollisionCheckTime > 0 and time - BBD.LastCollisionCheckTime < BBD.CollisionCheckInterval then return end
+    BBD.LastCollisionCheckTime = time
+
+    for door, _ in pairs( BBD.NonSolidDoors ) do
+        ---@cast door BBD.Door
+        if not IsValid( door ) then continue end
+
+        local collidingPlayers = BBD.GetCollidingPlayers( door )
+
+        -- If no players are colliding with the door, make it solid
+        if #collidingPlayers <= 0 then
+            door:SetCollisionGroup( COLLISION_GROUP_NONE )
+            BBD.NonSolidDoors[ door ] = nil
+            door:SetIsDoorSolidifying( false )
+
+            BBD.PlayRespawnSound( door )
+        end
+    end
+end
+
+-- Called when a door is respawned.
+---@param door BBD.Door Door that respawned
+---@param isPropBreach boolean Whether or not the door was prop breached
+BBD.RespawnDoor = function( door, isPropBreach )
+    if not IsValid( door ) then return end
+
+    -- Reset the door's rotation to its normal open position
+    local dirName = door:GetDamageDirection() == BBD.OPEN_DIRECTION_FORWARD and "m_angRotationOpenForward" or "m_angRotationOpenBack"
+    local resetRotation = door:GetInternalVariable( dirName )
+    local resetAngle = Angle( resetRotation.x, resetRotation.y, resetRotation.z )
+    if not IsValid( door:GetParent() ) then
+        -- DarkRP overrides ENT:SetAngle, so we use ENT:SetLocalAngles instead if the door doesn't have a parent
+        -- Hopefully they don't decide to override that too in the future...
+        door:SetLocalAngles( resetAngle )
+    else
+        door:SetAngles( resetAngle )
+    end
+
+    -- Reset the door's solidity to whatever it was before it was breached
+    door:SetSolid( BBD.PreBreachSolidity[ door ] )
+    BBD.PreBreachSolidity[ door ] = nil
+
+    -- Make sure the door isn't still trying to move
+    door:SetLocalAngularVelocity( ANGLE_ZERO )
+
+    -- Don't retain bullet holes and such when respawning
+    door:RemoveAllDecals()
+
+    -- Reset the door's health to the max health
+    door:SetHealthAfterLastDamage( conVarMaxHealth:GetFloat() )
+
+    if isPropBreach then
+        local propDoor = door:GetPropDoor()
+        if IsValid( propDoor ) then
+            propDoor:Remove()
+        end
+
+        if #BBD.GetCollidingPlayers( door ) > 0 then
+            -- Make the door non-solid to avoid trapping the player
+            door:SetCollisionGroup( COLLISION_GROUP_PASSABLE_DOOR )
+            door:SetIsDoorSolidifying( true )
+
+            -- Start checking to see if the player leaves the door's space
+            BBD.NonSolidDoors[ door ] = true
```

Comment:
I see the goal here. Can you have any kind of touch/untouch callbacks between a player and a door that could remove the re-check requirement?

Reply:
Players only trigger touch when the entity is solid, right?

Reply:
No clue honestly